pkg/logging/logger.go

```diff
@@ -59,13 +62,13 @@ func SetLogLevel(logLevel LogLevel) {
 // GetLogger returns a log producer with the given name
 func GetLogger(name string) OptimizelyLogProducer {
 	return NamedLogProducer{
-		name: name,
+		fields: map[string]interface{}{"name": name},
```

Comment:
Open to alternative suggestions to `"name"` as the key here. 

Perhaps `"package"` so we wouldn't use the `[Optimizely]` tag that we have in the default consumer, but instead use the actual package that is logging: (e.g. `"optimizely/go-sdk/pkg/client"`, `"optimizely/go-sdk/pkg/config/datafileprojectconfig"`, etc.)

Currently we don't use names that directly map to the struct or package. For example:
```go
var cmLogger = logging.GetLogger("PollingConfigManager")
```
The above name doesn't exist anywhere, only in logs.

Which in this proposal would be:
```go
var logger = logging.GetLogger("optimizely/go-sdk/pkg/config") 
```
since this is the package name that is logging.

This could simplify things as each package we'd have a single logger, but if this is not granular enough we can add the file name.
```go
var cmLogger = logging.GetLogger("optimizely/go-sdk/pkg/config/polling_manager") 
```

Reply:
I like this! It's way easier to map where the log messages came from.

Reply:
Cool, I'll create a follow-up.